File: data/src/main/java/org/oppia/data/backends/gae/model/WrittenTranslations.java
package org.oppia.data.backends.gae.model;

import com.squareup.moshi.Json;

import java.util.Map;

public class WrittenTranslations {

  /*
   * Parent Hierarchy: ExplorationContainer -> Exploration -> State -> WrittenTranslations
   */

  //TODO: ImmutableMap<> and Map<> error
  //ImmutableMap<> does not work but Map<> works
  //TODO: AutoJSON null data error

  @Json(name = "translations_mapping")
  public Map<String, Map<String, WrittenTranslation>> translationsMapping;

}

File: data/src/main/java/org/oppia/data/backends/gae/model/WrittenTranslations.kt
package org.oppia.data.backends.gae.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class WrittenTranslations(

  @Json(name = "translations_mapping") val translationsMapping: Map<String, Map<String, WrittenTranslation>>

)
